File: aspnet-in-practice/08-09/CS/CoolMVBlog/Controllers/TimeController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using Microsoft.Web.Mvc.AspNet4;

namespace CoolMvcBlog.Controllers
{
    public class TimeController : Controller
    {
        public ActionResult Index()
        {
            ViewData["ParentActionTime"] = DateTime.Now.ToLongTimeString();
            return View();
        }

        [OutputCache(Duration=30, VaryByParam="none")]
        [ChildActionCache(Duration=30)]
        public ActionResult CurrentServerTime()
        {
            ViewData["time"] = DateTime.Now.ToLongTimeString();

            return this.View();
        }
    }
}

File: aspnet-in-practice/08-09/VB/CoolMVCBlog/Controllers/TimeController.vb
Imports Microsoft.Web.Mvc.AspNet4

Namespace CoolMvcBlog
    Public Class TimeController
        Inherits System.Web.Mvc.Controller

        Public Function Index() As ActionResult
            ViewData("ParentActionTime") = DateTime.Now.ToLongTimeString

            Return View()
        End Function

        <OutputCache(Duration:=30, VaryByParam:="none")>
        <ChildActionCache(Duration:=30)>
        Public Function CurrentServerTime() As ActionResult
            ViewData("time") = DateTime.Now.ToLongTimeString
            Return View()
        End Function
    End Class
End Namespace